JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/GroupBean.java

```diff
@@ -146,31 +136,160 @@ private int getUserId(String userNickName) {
 
     private User getUser(String userNickName) {
 
-        User user = (User)em.createQuery("SELECT p FROM User p WHERE p.nickName=:nickname")
-                .setParameter("nickname", userNickName)
-                .getSingleResult();
+        User user;
+        try {
+            user = (User) em.createQuery("SELECT p FROM User p WHERE p.nickName=:nickname")
+                    .setParameter("nickname", userNickName)
+                    .getSingleResult();
+        }catch(NoResultException noResult) {
+            logger.log(Level.WARNING, "NO RESULT IN QUERY IN GETACCESS()", noResult);
+            return null;
+        }
 
         return user;
     }
 
-    private Access getAccess(String accessRights) {
+    public Access getAccess(String accessRights) {
+        Access access;
 
-        Access access = (Access)em.createQuery("SELECT p.accessId FROM Access p WHERE p.accessName =:accessrights")
-                .setParameter("accessrights", accessRights)
-                .getSingleResult();
+        try {
+            access = (Access) em.createQuery("SELECT p FROM Access p WHERE p.accessName =:accessname")
+                    .setParameter("accessname", accessRights)
+                    .getSingleResult();
+        }catch(NoResultException noResult) {
+            logger.log(Level.WARNING, "NO RESULT IN QUERY IN GETACCESS()", noResult);
+            return null;
+        }
         return access;
     }
 
-    private Group getGroup(int groupId) {
+    public Group getGroup(int groupId) {
+
+        Group group = null;
+        try {
+            group = (Group) em.createQuery("SELECT g FROM Group g WHERE g.groupId=:groupid")
+                    .setParameter("groupid", groupId)
+                    .getSingleResult();
+        }catch(NoResultException noResult) {
+            logger.log(Level.WARNING, "NO RESULT IN QUERY IN GETGROUP()", noResult);
+            return null;
+        }
+
+        return group;
+    }
+
+    public Group getGroup(String groupName) {
+
+        Group group = null;
+        try {
+            group = (Group) em.createQuery("SELECT g FROM Group g WHERE g.groupName=:groupname")
+                    .setParameter("groupname", groupName)
+                    .getSingleResult();
+        }catch(NoResultException noResult) {
+            logger.log(Level.WARNING, "NO RESULT IN QUERY IN GETGROUP()", noResult);
+            return null;
+        }
+        return group;
+    }
+
+    public int getGroupId(String groupName) {
```

Comment:
I had to create this method for test, but it can be deleted.

23 марта 2016 г., в 2:14, alexander1leonov <notifications@github.com> написал(а):

> In JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/GroupBean.java:
> 
> > +
> > +    public Group getGroup(String groupName) {
> > +
> > +        Group group = null;
> > +        try {
> > +            group = (Group) em.createQuery("SELECT g FROM Group g WHERE g.groupName=:groupname")
> > +                    .setParameter("groupname", groupName)
> > +                    .getSingleResult();
> > +        }catch(NoResultException noResult) {
> > +            logger.log(Level.WARNING, "NO RESULT IN QUERY IN GETGROUP()", noResult);
> > +            return null;
> > +        }
> > +        return group;
> > +    }
> > +
> > +    public int getGroupId(String groupName) {
> You do not need this method. getGroup(String) is enough.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>